week3/homework/src/actions/markAsDone.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+async function markAsDone(todo, request, response) {
+  try {
+    let todoList = await todo.list().then(todoList => JSON.parse(todoList));
+    const index = parseInt(request.params.id);
+    if (todoList[index].done === true) {
```

Comment:
You may need to check if the ID provided by user exists

Reply:
Also done... thanks